### DISCUSSION TITLE
Data storage

### DISCUSSION THREAD

**Discussion by aomayden** [AltimateAI/vscode-dbt-power-user#1408] [Q&A]
Hi,

I was wondering whether this extension processes data on an external server owned by Altimate.ai or whether all processing is done on my own machine within the vscode extension?

Thanks!

**Comment by anandgupta42** [comment]
@kernowalmayden By data, I am assuming you mean meta-data. Some of the features that require the API key interact with the Altimate AI backend or else the rest of the features work locally. You can find more details here: https://docs.myaltimate.com/arch/faq/
